handlers/common.go

```diff
@@ -31,10 +32,58 @@ func PingNoOp(event *events.Event, apiClient *client.RancherClient) error {
 	return nil
 }
 
+func acquire_networked_global_persistent_lock(resourceId string) error {
+	return nil
+}
+
+func release_lock(lock string) {
+	return
+}
+
 func getMachineDir(machine *client.Machine) (string, error) {
 	return getBaseMachineDir(machine.ExternalId)
 }
 
+func sync_resource(machine *client.Machine) error {
```

Comment:
@wlan0 I had an idea:
what if we _always_ recreated the machine directory based off of the `extractedConfig` field on the resource, regardless of what is on disk? This could prevent us from running into issues where the data on disk is stale compared to what is in cattle and make us more resilient to partial corruptions (like a single file getting delete (not that I expect that to happen)).

Basically, it would mean that we _always_ go to cattle as the source of truth and that the files on disk are just meaningful for the duration of the docker-machine operation.

Based on previous conversations, I think that you'll be hesitant because of the extra overhead of getting and extracting the config on every event, which is a valid concern. My opinion there is that it is worth the extra overhead to get the extra stability and predictability.